x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -549,6 +573,76 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             }
         }
 
+        try {
+            // all certs and keys have been generated in the temp certs dir, therefore:
+            // 1. backup (move) any previously existing tls certs dir (this backup is NOT removed when auto-conf finishes)
+            if (Files.exists(env.configFile().resolve(TLS_GENERATED_CERTS_DIR_NAME))) {
```

Comment:
If it happens that there's already a `certs` dir present, we back it up and never remove it.

Reply:
I don't have anything against doing this for sure, but the only way to get here is if we :

- have a certs directory 
- no configuration in elasticsearch.yml
- no entries in the keystore

so  I don't expect this happening that much , right ?

Reply:
Right, it is a corner case that the admin had such a config directory while the the yml and keystore files are clean. Not a reason strong enough to prevent auto-configuration.